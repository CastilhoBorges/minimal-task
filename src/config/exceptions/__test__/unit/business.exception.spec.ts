import { HttpStatus } from '@nestjs/common';
import { BusinessException } from '../../business.exception';

describe('BusinessException', () => {
  it('should create an instance of BusinessException with correct properties', () => {
    const message = 'Invalid business logic';
    const metadata = { code: 'INVALID_ACTION' };

    const exception = new BusinessException(message, metadata);

    expect(exception).toBeInstanceOf(BusinessException);
    expect(exception.getResponse()).toEqual({
      name: 'BusinessException',
      message,
      statusCode: HttpStatus.BAD_REQUEST,
      metadata,
      stack: undefined,
    });
    expect(exception.getStatus()).toBe(HttpStatus.BAD_REQUEST);
    expect(exception.getResponse()).toMatchSnapshot();
  });

  it('should create an instance with only message', () => {
    const message = 'Invalid business logic';

    const exception = new BusinessException(message);

    expect(exception).toBeInstanceOf(BusinessException);
    expect(exception.getResponse()).toEqual({
      name: 'BusinessException',
      message,
      statusCode: HttpStatus.BAD_REQUEST,
      metadata: undefined,
      stack: undefined,
    });
    expect(exception.getStatus()).toBe(HttpStatus.BAD_REQUEST);
    expect(exception.getResponse()).toMatchSnapshot();
  });
});
