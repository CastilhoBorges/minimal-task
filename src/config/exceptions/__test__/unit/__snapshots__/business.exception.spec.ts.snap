// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BusinessException should create an instance of BusinessException with correct properties 1`] = `
{
  "message": "Invalid business logic",
  "metadata": {
    "code": "INVALID_ACTION",
  },
  "name": "BusinessException",
  "stack": undefined,
  "statusCode": 400,
}
`;

exports[`BusinessException should create an instance with only message 1`] = `
{
  "message": "Invalid business logic",
  "metadata": undefined,
  "name": "BusinessException",
  "stack": undefined,
  "statusCode": 400,
}
`;
